pkg/prometheus/promcfg.go

```diff
@@ -81,14 +85,37 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 
 	logger = log.WithSuffix(logger, "version", promVersion)
 
+	scrapeClasses, defaultScrapeClass, err := getScrapeClassConfig(cpf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse scrape classes: %w", err)
+	}
+
 	return &ConfigGenerator{
 		logger:                 logger,
 		version:                version,
 		prom:                   p,
 		endpointSliceSupported: endpointSliceSupported,
+		scrapeClasses:          scrapeClasses,
+		defaultScrapeClass:     defaultScrapeClass,
 	}, nil
 }
 
+func getScrapeClassConfig(cpf monitoringv1.CommonPrometheusFields) (map[string]monitoringv1.ScrapeClass, string, error) {
+	scrapeClasses := map[string]monitoringv1.ScrapeClass{}
+	defaultScrapeClass := ""
+	for _, scrapeClass := range cpf.ScrapeClasses {
+		scrapeClasses[scrapeClass.Name] = scrapeClass
```

Comment:
In this line, we're copying the whole ScrapeClass object inside the map, since it isn't a pointer, and we do it once for each ScrapeClass. We could just copy the pointer address instead

At the moment ScrapeClass only has 2 configuration options, TLS and Authorization, but once we add relabelings... 💀

Reply:
For sure 😄